src/Storage/Controllers/InstancesController.cs

```diff
@@ -276,6 +276,35 @@ public async Task<ActionResult<Instance>> Get(int instanceOwnerPartyId, Guid ins
             }
         }
 
+        /// <summary>
+        /// Gets process info relevant to authorization
+        /// </summary>
+        /// <param name="instanceOwnerPartyId">The party id of the instance owner.</param>
+        /// <param name="instanceGuid">The id of the instance to retrieve.</param>
+        /// <returns>Authorization info.</returns>
+        [HttpGet("AuthInfo/{instanceOwnerPartyId:int}/{instanceGuid:guid}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Produces("application/json")]
+        [ApiExplorerSettings(IgnoreApi = true)]
+        public async Task<ActionResult<AuthInfo>> GetForAuth(int instanceOwnerPartyId, Guid instanceGuid)
+        {
+            try
+            {
+                (Instance instance, _) = await _instanceRepository.GetOne(instanceOwnerPartyId, instanceGuid, false);
+                if (instance.InstanceOwner.PartyId != instanceOwnerPartyId.ToString())
+                {
+                    throw new Exception("Party id not found");
```

Comment:
Throwing a new exception for the sole purpose of breaking out of the code block should be avoided.

I can suggest two alternatives:
1. return NotFound().
2. Invert the if and adjust logic to match.

Reply:
Done